pyvista/core/_validation/_cast_array.py

```diff
@@ -103,10 +103,11 @@ def _cast_to_numpy(arr, /, *, as_any=True, dtype=None, copy=False, must_be_real=
             if copy:
                 out = out.copy()
         else:
-            out = np.array(arr, dtype=dtype, copy=copy)
+            # handle copy flag manually because numpy>=2.0 raises if np.array(..., copy=False) must copy
+            out = np.array(arr, dtype=dtype) if copy else np.asarray(arr, dtype=dtype)
```

Comment:
This is too ugly for my taste, but I didn't find a more elegant solution:
1. `np.array(..., copy=False)` raises on old NumPy if copying fails
2. `np.array(..., copy=None)` which doesn't raise doesn't exist on old NumPy
3. `np.asarray(..., copy=False)` doesn't exist on old NumPy.

So I think there's no one function call we can use here.

Reply:
`np.asarray` only copies if needed for both NumPy 1.26 and 2.0. So, you could probably refactor the whole function instead, and only copy explicitly at the end if needed. Something like:

``` python
# Do casting
out = np.asanyarray(arr, dtype=dtype) if as_any else np.asarray(arr, dtype=dtype)

# Array may have been copied by the casting. Only copy if necessary.
if copy and out is arr:
    out = out.copy()
```

Reply:
Done in https://github.com/pyvista/pyvista/pull/6156/commits/62d915a2a0f5227a4bc518eedcbf5b5a0354753c.